src/transformStateWithClones.js

```diff
@@ -8,6 +8,28 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newState = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+    }
```

Comment:
A `default` case should be added to the `switch` statement to handle any unexpected `action.type`. This is important for error handling and to ensure that the function behaves predictably even with incorrect input.